fenicsprecice/fenicsprecice.py

```diff
@@ -294,8 +345,8 @@ def initialize(self, coupling_subdomain, mesh, function_space, write_function=No
                                 "No proper treatment for dimensional mismatch is implemented. Aborting!".format(
                     self._fenics_dimensions, self._interface.get_dimensions()))
 
-        fenics_vertices, self._coupling_mesh_vertices = get_coupling_boundary_vertices(
-            mesh, coupling_subdomain, self._fenics_dimensions, self._interface.get_dimensions())
+        fenics_vertices, self._coupling_mesh_vertices = get_coupling_boundary_vertices(self._read_function_space.mesh(),
+                coupling_subdomain, self._fenics_dimensions, self._interface.get_dimensions())
```

Comment:
What happens, `if read_function_space is None`? I would rather simply use a `mesh` here that has been initialized previously according to the `CouplingMode`.

Reply:
I think this also improves readability.